platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -0,0 +1,1027 @@
+   2. remove the previous candidate roster hash from the voting closed set.
+
+`On TssMessageTransaction`:
+
+1. If the voting is closed, do nothing.
+2. If a `TssVoteTransaction` exists for this node, do nothing.
+3. If a TSS Message already exists for the share index and target roster, do nothing.
+4. Validate the TssMessage and if valid, add it to the list of TssMessageTransactions for the target roster.
+5. Update the sequence number bit in the vote vector with the results of validation for the TssMessageTransaction.
+6. If there are (active roster) threshold number of valid TssMessageTransactions for the target roster, send
+   the valid TssMessageTransactions transcript to the TSS Key Manager
+
+`On TssVoteTransaction`:
+
+1. If voting is closed or the vote is a duplicate of an existing vote, do nothing.
+2. Add the vote to the total and check if the threshold is met.
+3. If the threshold is met, add the roster hash to the voting closed set.
+
+`On PublicSharesAndLedgerId`:
+
+1. If the public shares and ledger id are for the candidate roster,
+   1. Sign the ledger ID with the node's private signing key.
+   2. send a `TssVoteTransaction` with the ledger id and the vote vector.
+
+##### Validate TSS Message List
+
+1. For the votes already cast in the vote vector
+   1. sum all the yes votes into a total count.
+   2. If the count is greater than or equal to the threshold in the candidate roster, do nothing and return.
+2. For each TSS message in the TSS Message list
+   1. If the TSS Message has a yes entry in the vote vector, do nothing.
+   2. otherwise, validate the message.
+   3. If the message is valid, update the vote vector and increment the count of yes votes.
+   4. If the count of yes votes is greater than or equal to the threshold in the candidate roster, then send the
+      vote vector as a system transaction and exit the validation process.
+
+#### TSS Key Manager
+
+The TSS Key Manager is responsible for generating the public and private keys of the shares for a fully keyed roster.
+It receives on input the bundle of TssMessages that generate share keys and produces on output the private shares
+for the node, the public shares, and the ledger id.
+
+Internal Elements:
+
+1. Map<RosterHash, Roster> rosters
+2. Map<RosterHash, List<TssMessage>> tssMessages
+3. Map<RosterHash, Record(RosterHash, List<PairingPrivateKey>, List<PairingPublicKey>, LedgerId)> shares
+
+Inputs:
+
+1. Pair<RosterHash, Roster> (Wire)
+2. Pair<RosterHash, List<TssMessage>> (Wire)
+
+Outputs
+
+1. Record(RosterHash, List<PairingPrivateKey>, List<PairingPublicKey>, LedgerId) (Wire)
+
+#### Transaction Resubmitter
+
+The transaction resubmitter is extended to detect which the new `TssMessageTransaction` and `TssVoteTransaction`
+system transactions have not reached consensus because the event containing them has become stale. The Transaction
+Resubmitter will resubmit these transactions to the transaction pool for reprocessing.
```

Comment:
```suggestion
The transaction resubmitter is extended to detect which of the new `TssMessageTransaction` and `TssVoteTransaction`
system transactions have not reached consensus because the event containing them has become stale. The transaction
resubmitter will resubmit these transactions to the transaction pool for reprocessing.
```